xbmc/addons/Repository.cpp

```diff
@@ -300,7 +300,17 @@ CRepository::DirInfo CRepository::ParseDirConfiguration(const CAddonExtensions&
     }
   }
 
-  dir.version = AddonVersion{configuration.GetValue("@minversion").asString()};
+  std::string minversion = configuration.GetValue("@minversion").asString();
+  std::string maxversion = configuration.GetValue("@maxversion").asString();
+
+  if (minversion.empty())
```

Comment:
Instead of the magic numbers why not remove these and instead do the `empty()` checks as part of the condition above?

Reply:
i had commented about this on my 2nd reply here

"the setting maxversion to "999.9.9" is ugly, but it's the easiest way and easy to understand.

Other way would be to check maxversion isn't 0.0.0 and then compare it.
if (version >= dir.minversion && (dir.maxversion == AddonVersion('0.0.0') || version < dir.maxversion))
Not so easy to understand.

Also, setting minversion = "0.0.0" isn't actually required as AddonVersion("") returns 0.0.0
However, I added it in to make it clearer and more explicit."

Reply:
True. But the empty checks are explicit and the meaning is obvious. Less digging than assuming the 0.0.0 value for minversion.

Reply:
the empty checks won't work with an AddonVersion object though. Only on the strings.
Once that empty string (what is returned if no value is given) is given to dir.minversion = AddonVersion{minversion};
dir.minversion is now AddonVersion object with "0.0.0" as it's version.
You can't do empty on dir.minversion as 1) its AddonVersion object (not a string) 2) it's set to '0.0.0' so not empty.

Reply:
So you'd need` if (version >= dir.minversion && (dir.maxversion == AddonVersion('0.0.0') || version < dir.maxversion))`
But that's even more ugly & less obvious. 
It would also break if AddonVersion was updated to not set a default version of '0.0.0'.

Reply:
Correct

Reply:
If the member gets initialized to 0.0.0 when you instanciate an addonversion object with an empty string why not implement `isempty` method on the class to hide the implementation details from the consumer?
It's pretty much the same as you have now, but at least you avoid the 0.0.0 and 999.9.9 "hack"

Reply:
If the member gets initialized to 0.0.0 when you instanciate an addonversion object with an empty string why not implement `isempty` method on the class to hide the implementation details from the consumer?
It's pretty much the same as you have now, but at least you avoid the 0.0.0 and 999.9.9 "hack"

Edit: looks like empty is already there

Reply:
i wasn't aware AddonVersion.empty existed :) That will work fine in the check as @phunkyfish  said.
I'll push update shortly!